src/sql/base/browser/ui/chart/chart.component.ts

```diff
@@ -0,0 +1,321 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { Component, Inject, forwardRef, ChangeDetectorRef } from '@angular/core';
+import * as chartjs from 'chart.js';
+import { mixin } from 'sql/base/common/objects';
+import { Disposable } from 'vs/base/common/lifecycle';
+import * as azdata from 'azdata';
+
+export interface BarDataSet {
+	label: string;
+	data: number[];
+	backgroundColor?: string[];
+	borderColor?: string[];
+}
+
+@Component({
```

Comment:
It's fine to leave it like this for this one, but in the future it'd be better to create a "native" UI component (using just the lib and DOM manipulation stuff), and then just have the modelview angular component use that. The less angular we use for the core components the better.

Reply:
Changing to use less angular sounds good by me; I'll make a note to swap that in a follow-up PR if it doesn't make this one.  Since it's working and I'd have to figure out how do to the change-over, I want to focus on getting this in and having a clean slate to change that from.

Reply:
This was more of a "if you make a new component in the future keep this in mind". Converting will be a bunch of work - and given that we already have a bunch of other components that are done this way it's fine to keep it like this until we have a reason to specifically change it.